gradle/.qa/checkstyle.gradle

```
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+project.apply plugin: "checkstyle"
+
+checkstyle {
+    configDirectory = rootProject.layout.projectDirectory.dir("checkstyle")
+    configProperties = [importControlFile: "import-control.xml"]
+    toolVersion = versions.checkstyle
+}
+
+configure(checkstyleMain) {
+    group = 'Verification'
+    description = 'Run checkstyle on all main Java sources'
+}
+
+configure(checkstyleTest) {
+    group = 'Verification'
+    description = 'Run checkstyle on all test Java sources'
+}
+
+test.dependsOn(checkstyleMain, checkstyleTest)
```

```suggestion
check.dependsOn(checkstyleMain, checkstyleTest)
```
"normally" not seen being bound to the `test` goal. 

- https://github.com/apache/kafka/pull/20219/files#r2244732670

like done in junit:

- https://github.com/junit-team/junit-framework/pull/4708
- https://github.com/junit-team/junit-framework/pull/4708/files#diff-b6c0ec6facb84213a6f79b874524c6093403e575490d547a04ac8e6102d33cbaR317